CorsixTH/Lua/app.lua

```diff
@@ -1839,16 +1839,32 @@ function App:checkForUpdates()
   end
 
   local update_table = loadstring_envcall(update_body, "@updatechecker") {}
+  update_table.revision = update_table.revision or 0
   local changelog = update_table["changelog_" .. default_language]
-  local new_version = update_table["major"] .. '.' .. update_table["minor"] .. update_table["revision"]
-
-  if (new_version <= current_version) then
+  local new_version = update_table.major .. '.' .. update_table.minor .. '.' .. update_table.revision
+
+  -- Semantic version comparison of the current and update version numbers
+  local function semver()
+    local current_major, current_minor, current_revision = string.match(current_version, "(%d+)%.(%d+)%.?(%d*)")
+    current_major, current_minor = tonumber(current_major), tonumber(current_minor)
+    if current_major > update_table.major then return true
+    elseif current_major < update_table.major then return false
+    end
+    if current_minor > update_table.minor then return true
+    elseif current_minor < update_table.minor then return false
+    end
+    current_revision = tonumber(current_revision) or 0
+    if current_revision >= update_table.revision then return true
+    else return false
+    end
+  end
+  if semver() then
```

Comment:
Is the function name correct? Semver what you're using, but the comment needs to tell me you're doing a compare. Whereas, it's likely going to be better to communicate it in the function name.

You could also split this out into two functions. One to convert to semver and another to compare versions. It may also be possible then to make this a little less messy.